spec/memoize_query_spec.rb

```diff
@@ -612,4 +537,246 @@ def expect_to_eq_with_or_without_redis
+          expect_to_use_redis do
+            expect(relation.reload.to_a).to eq(result)
+          end
+        end
+      end
+    end
+
+    RSpec.shared_examples 'with local cache' do
+      it 'only invalidates the affected query result sets' do
+        RedisMemo::Cache.with_local_cache do 
+          relations_with_comparator_and_other.each do |relation|
+            expect_to_use_redis do
+              expect(relation.to_a).to eq(unaffected_result)
+            end
+          end
+  
+          unaffected_creation
+  
+          # Does not affect with the creation because it is not dependent
+          relations_with_comparator_and_other.each do |relation|
+            expect_not_to_use_redis do
+              expect(relation.reload.to_a).to eq(unaffected_result)
+            end
+          end
+  
+          update
+  
+          # Will affect with the update
+          relations_with_comparator_and_other.each do |relation|
+            expect_to_use_redis do
+              expect(relation.reload.to_a).to eq(affected_result)
+            end
+          end
+        end
+      end
+    end
+  end
+
+  context 'when the queries have greater than (and or equal) operators' do
+    let!(:record) { Site.create!(a: 2, b: 2) }
+    
+    # The following two relations are both SELECT * FROM Site where a > 1
+    let!(:relation1_with_only_comparator) { Site.where('a > ?', 1) }
+    let!(:relation2_with_only_comparator) { Site.where(a: 2..Float::INFINITY) }
+    let!(:relation3_with_only_comparator) { Site.where('a > ? AND b = ?', 1, 2) }
+    # The following two relations are both SELECT * FROM Site where a >= 2
+    let!(:relation4_with_only_comparator) { Site.where('a >= ?', 2) }
+    let!(:relation5_with_only_comparator) { Site.where('a >= ? AND b = ?', 2, 2) }
+    let!(:relations_with_only_comparator) do
+      [
+        relation1_with_only_comparator, 
+        relation2_with_only_comparator, 
+        relation3_with_only_comparator,
+        relation4_with_only_comparator,
+        relation5_with_only_comparator
+      ] 
+    end
+
+    # The following two relations are both SELECT * FROM Site where a > 1 AND b = 2
+    let!(:relation1_with_comparator_and_other) { Site.where('a > ?', 1).where(b: 2) }
+    let!(:relation2_with_comparator_and_other) { Site.where(a: 2..Float::INFINITY).where(b: 2) }
+    # The following two relations are both SELECT * FROM Site where a >= 2 AND b = 2
+    let!(:relation3_with_comparator_and_other) { Site.where('a >= ?', 2).where(b: 2) }
+    let!(:relations_with_comparator_and_other) do
+      [
+        relation1_with_comparator_and_other, 
+        relation2_with_comparator_and_other, 
+        relation3_with_comparator_and_other,
+      ] 
+    end
+
+    # Deliberately use let below to let it be lazy-evaluated
+    # An update which will affect the results of the relation_with_comparator_and_other above
+    let(:update) { record.update(a: 1) }
```

Comment:
You probably want this to be a function. Currently it's just a variable named `update`

Reply:
this has been removed in new commit